loadfile/loadfile.go

```diff
@@ -0,0 +1,47 @@
+// Package loadfile provides functions to load files from a directory.
+package loadfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// LoadContext reads the contents at each file into a map where the key
+// is the absolute filepath and file contents is the value.
+func LoadContext(neededFilepaths []string) (map[string][]byte, error) {
+	result := map[string][]byte{}
+	var err error
+	for _, filePath := range neededFilepaths {
+		absPath, err := filepath.Abs(filePath)
+		if err != nil {
+			return nil, fmt.Errorf("error loading context files, failed to obtain absolute path of file %s (%w)",
+				filepath.Base(filePath), err)
+		}
```

Comment:
Why do you report `filepath.Base(filePath)` instead of the actual failing input, `filePath`?

Reply:
This could result in awkward errors; for example I get that you don't want to report an error in `--workflow foo/bar.yaml` as `/var/tmp/whatever/foo/bar.yaml`, but I'm not sure that just `bar.yaml` is a useful improvement.

Then again, this should be a highly unlikely error as we already expanded any relative paths to `Abs` via the `Join`. (And in fact it would probably be better to detect and report *there* that the result of the `Join`, assuming that `Join` doesn't care, isn't a legal absolute path.)

Reply:
yeah, I didn't want to report the whole filepath since it is probably quite long. I can change it to report the absolute filepath.

Reply:
> I didn't want to report the whole filepath since it is probably quite long.

The question is, does that length represent a problem?  On the other side of the trade-off is the fact that brevity might be omitting details which are necessary for the user to diagnose and resolve the problem.  Unless you're really sure that the user would know where to find the file, you really should offer the whole path -- the extra 50-60 characters to the message probably won't cause a problem anything like the one posed by not knowing which directory the file is in.